### DISCUSSION TITLE
Multiple (duplicate) "Dependency Dashboard" issues created.

### DISCUSSION THREAD

**Discussion by undergroundwebdesigns** [renovatebot/renovate#12131] [Request Help]
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### Describe the bug

I added Renovate to a new (private) repository yesterday, and configured it by adding `"dependencyDashboardApproval": true` and an "extends" call to a local config that defines some HTTP credentials for private repositories. At the time I set up Renovate, the repository had issues _disabled_, and it was only a couple hours later when I went looking for the Renovate "Dependency Dashboard" issue that I realized this, and at that point I enabled issues for the repo.

Since that time, Renovate has been creating a new "Dependency Dashboard" issue approximately once per hour, each time the job runs (as shown by the logs on https://app.renovatebot.com/). The issues are identical as best I can tell, and contain the information I would expect. There's nothing interesting in the logs except a note that the `ensureIssue` method is indeed creating a new issue each run.

I attempted to create a reproduction repository following the same procedure (starting with issues enabled, etc) with a much simpler repo / dependency file and was unable to reproduce the behaviour.

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Ensuring Dependency Dashboard
DEBUG: ensureIssue(Dependency Dashboard)
INFO: Issue created
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
Can you share a job ID from one of the runs?

**Comment by undergroundwebdesigns** [comment]
Sure, as of right now the most recent three runs are 476457705, 476405074, and 476363944. All of them created new (duplicate) "dependency dashboard" issues in my repo.

**Comment by rarkins** [comment]
Looking at the most recent job in that repo, this early log is probably pointing to the root cause:

```
{"level":20,"msg":"findIssue(Dependency Dashboard)","time":"2021-10-08T04:45:51.033Z"}
{"level":20,"msg":"Retrieving issueList","time":"2021-10-08T04:45:51.033Z"}
{"level":20,"msg":"Retrieved 0 issues","time":"2021-10-08T04:45:51.241Z"}
```

For some reason, Renovate is finding no issues in the repo.

https://github.com/renovatebot/renovate/blob/a9c9aa02d8ae900e651a4e0051c834249631dddf/lib/platform/github/index.ts#L1001

then https://github.com/renovatebot/renovate/blob/a9c9aa02d8ae900e651a4e0051c834249631dddf/lib/platform/github/index.ts#L979-L990

Until we can work out the full root cause, unfortunately the best workaround is to disable the dashboard.

Can you perform this troubleshooting?

1. Go to https://docs.github.com/en/graphql/overview/explorer
2. Paste in this query:
```graphql
query {
  repository(owner: "renovatebot", name: "renovate") {
    issues(
      orderBy: { field: UPDATED_AT, direction: DESC },
      filterBy: { createdBy: "renovate[bot]" },
      first: 3
    ) {
      pageInfo {
        endCursor
        hasNextPage
      }
      nodes {
        number
        state
        title
        body
      }
    }
  }
}
```
(but change the owner + name values on the second line to match with your repo)
3. See if any results are found

**Comment by undergroundwebdesigns** [comment]
Well, that did help identify the issue. Somehow all the renovate issues are being created as being owned by one of our internal Github service accounts, and not `renovate[bot]`. The query above gives no results unless I remove the "createdBy" filter. Now that I know what to look for, you can even see that in the GH UI:
![Screen Shot 2021-10-12 at 8 58 09 AM](https://user-images.githubusercontent.com/405377/136960719-074e3c5b-9c8a-4d38-b725-02f26850f251.png)

I've just tried removing renovate from this repo and re-adding it to see if that changes anything.

**Reply by rarkins** [reply]
If you have someone in your company running their own renovate bot then uninstalling or reinstalling the Renovate app isn't going to help at all

**Comment by robgordon89** [comment]
I also have this issue, I think it is related to how the issue is created, my issue's are created by a github user (not renovate) as we have to use hostRules to setup auth for github.com

The query above returns no results, and so renovate automatically creates a new dashboard each time.

**Reply by HonkingGoose** [reply]
Quote from [our docs on authentication](https://docs.renovatebot.com/getting-started/running/#authentication):

> ### Authentication
> 
> Regardless of platform, you need to select a user account for `renovate` to assume the identity of, and generate a Personal Access Token.
> We recommend you use `@renovate-bot` as username if you're on a self-hosted server where you can set all usernames.
> We also recommend you configure `config.gitAuthor` with the same identity as your Renovate user, e.g. like `"gitAuthor": "Renovate Bot <renovate@whitesourcesoftware.com>"`.
> 
> 
> !!! warning
>     We recommend you use a single, dedicated username for your Renovate bot.
>     Never share the Renovate username with your other bots, as this can cause flip-flopping.

**Comment by hskrtich** [comment]
I was running into these same issues. The problem turned out that since I was trying to use the github hosted app I *shoudln't* be setting auth in the config. I just needed to make sure that the app had access to the private repo and remove the hostRule based auth. 

Then PR's, Issues and branches were all being created by the standard renovate[bot] user and everything worked as expected.

**Reply by robgordon89** [reply]
the issue with this is that if you need hostRule for other private repos in GitHub then you will always hit this issue.

The lookup of the dashboard should be done via a label rather than the user that created it I think 🤔 

I still can't get this to work with the dashboard, and without the dashboard, I can't force PR's, etc it is really frustrating

Anyone else got any ideas on how to fix this?

I have tried changing the renovateUsername and other config options to my bot user to no avail.

**Comment by robgordon89** [comment]
Im also running into another issue now that is causing issues with PR's 

When we close a PR (as we don't need to update) renovate will just reopen the same PR again, I think it has something to do with the user/bot that created the PR etc 

This is super annoying, this only happens when you have a hostRule for private Github repos, etc.

Logs:
```
DEBUG: getBranchPr(renovate/docker.io-timescale-timescaledb-2.7.x)(branch="renovate/docker.io-timescale-timescaledb-2.7.x")
DEBUG: findPr(renovate/docker.io-timescale-timescaledb-2.7.x, undefined, open)(branch="renovate/docker.io-timescale-timescaledb-2.7.x")
DEBUG: findPr(renovate/docker.io-timescale-timescaledb-2.7.x, undefined, closed)(branch="renovate/docker.io-timescale-timescaledb-2.7.x")
DEBUG: Creating PR(branch="renovate/docker.io-timescale-timescaledb-2.7.x")
{
  "prTitle": "chore(deps): update dependency docker.io/timescale/timescaledb to v2"
}
DEBUG: Creating PR(branch="renovate/docker.io-timescale-timescaledb-2.7.x")
{
  "title": "chore(deps): update dependency docker.io/timescale/timescaledb to v2",
  "head": "...",
  "base": "develop",
  "draft": false
}
```

**Comment by odin568** [comment]
Running into same issue. GitLab requires yearly rollover of access token. With an access token, an artifical user is created. The name is derived from the token name, but technically has another guid.  
Now after token renwal, all projects have a new dependency dashboard :(   
Also looked through documentation, no real solution so far still, right?

**Reply by rarkins** [reply]
Please start a new discussion, and provide more details than just "access token".
